### PR TITLE
Feature: Rename Item

### PR DESCRIPTION
**Pull request by YehorPytomets**
### Changes introduced by this PR:
* Implement feature of modifying file names.
* Edit folder name by producing two single clicks.
* Submit changes by pressing `Enter`.
* Changed name is visible only when server responded to it.

### CODE REVIEW

**Comment by YehorPytomets** [comment]
Left to do:
- [ ] Fix issue below.
- [ ] Add tests.

**Comment by YehorPytomets** [comment]
@dkalinin , PR is ready to review.

File: webclient/app/pages/file-list-explorer.js [hunk 1]
```diff
@@ -120,14 +121,23 @@ export default class FileListExplorer extends StateAwareComponent {
     });
```
Comment (dkalinin) [inline webclient/app/pages/file-list-explorer.js:121]: - [ ] During initialization list component should have all properties that are available from state. We need to read state and pass proper properties to FIleList component.

File: webclient/app/components/file-list/file-list.js [hunk 2]
```diff
@@ -34,7 +46,73 @@ export default class FileList extends Component {
...
+
+      itemComponent.onInputCommit((value) => {
+        const updatedItem = Object.assign({}, listItemModel);
+
+        if (!value || updatedItem.name === value) {
+          return;
+        }
+
+        updatedItem.name = value;
+        this._onUpdateListItemHandlers.executeHandlers(updatedItem);
```
[the remark this answers is not in the document]
Comment (YehorPytomets) [reply webclient/app/components/file-list/file-list.js]: Done.

**Comment by YehorPytomets** [comment]
Fix issue #52 
Cause: Because of _rerendering of ListItemComponent we had multiple calls to addEventListeners() that registered the same listeners multiple times. So, when we produced one of such events we had it's listener executed multiple time. Moreover, some of such events triggered FileList's code, which registered this listeners multiple times again, thus making things worse.
Solution: Now component has mechanism of registering dom event listeners which get cleared each time _rerender() is called. This asserts that no more than expected number of listeners are registered for given element at the time.

**Comment by YehorPytomets** [comment]
@dkalinin , PR is ready to review.

**Comment by YehorPytomets** [comment]
@dkalinin , PR is ready to review.

**Pull request merged**